plugins/runner/internal/runtimehandler/binary/handler.go

```diff
@@ -0,0 +1,202 @@
+	inputDirMountPoint   string
+	pluginServerEndpoint string
+	ready                bool
+}
+
+func New(ctx context.Context, config types.PluginConfig) (runtimehandler.PluginRuntimeHandler, error) {
+	return &binaryRuntimeHandler{
+		config: config,
+	}, nil
+}
+
+func (h *binaryRuntimeHandler) Start(ctx context.Context) error {
+	h.pluginDir = filepath.Join(os.TempDir(), "vmclarity-plugins", h.config.Name)
+
+	image, err := stereoscope.GetImage(ctx, h.config.ImageName)
+	if err != nil {
+		return fmt.Errorf("unable to get image(%s): %w", h.config.ImageName, err)
+	}
+
+	if _, err := os.Stat(h.pluginDir); os.IsNotExist(err) {
+		err = containerrootfs.ToDirectory(ctx, h.config.ImageName, h.pluginDir)
+		if err != nil {
+			return fmt.Errorf("unable to extract image(%s): %w", h.config.ImageName, err)
+		}
+	}
+
+	// Mount input from host
+	// /tmp/vmclarity-plugins/kics + /input + /host-dir-to-scan
+	h.inputDirMountPoint = filepath.Join(h.pluginDir, runtimehandler.RemoteScanInputDirOverride, h.config.InputDir)
+	err = os.MkdirAll(h.inputDirMountPoint, 0755)
+	if err != nil {
+		return fmt.Errorf("unable to create directory for mount point: %w", err)
+	}
+
+	// nolint:typecheck
+	err = syscall.Mount(h.config.InputDir, h.inputDirMountPoint, "", syscall.MS_BIND, "")
+	if err != nil {
+		return fmt.Errorf("unable to mount input directory (%s - %s): %w", h.config.InputDir, h.inputDirMountPoint, err)
+	}
+
+	// https://lwn.net/Articles/281157/
+	// "the read-only attribute can only be added with a remount operation afterward"
+	err = syscall.Mount(h.config.InputDir, h.inputDirMountPoint, "", syscall.MS_BIND|syscall.MS_REMOUNT|syscall.MS_RDONLY, "")
+	if err != nil {
+		return fmt.Errorf("unable to remount input directory as read-only (%s - %s): %w", h.config.InputDir, h.inputDirMountPoint, err)
+	}
+
+	// TODO: handle panics - unmount
+
+	// Determine entrypoint or command to execute
+	var args []string
+	if len(image.Metadata.Config.Config.Entrypoint) > 0 {
+		args = append(image.Metadata.Config.Config.Entrypoint[0:], image.Metadata.Config.Config.Cmd...)
+	} else if len(image.Metadata.Config.Config.Cmd) > 0 {
+		args = image.Metadata.Config.Config.Cmd[0:]
+	} else {
+		return fmt.Errorf("no entrypoint or command found in the config")
+	}
+
+	// Find a port
+	openPortListener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return fmt.Errorf("unable to find port")
+	}
+	port := openPortListener.Addr().(*net.TCPAddr).Port
+
+	h.pluginServerEndpoint = fmt.Sprintf("http://127.0.0.1:%d", port)
+
+	// Set environment variables
+	env := image.Metadata.Config.Config.Env
+	env = append(env, fmt.Sprintf("%s=%s", plugin.EnvListenAddress, h.pluginServerEndpoint))
+
+	// Set workdir
+	workDir := image.Metadata.Config.Config.WorkingDir
+	if workDir == "" {
+		workDir = "/"
+	}
+
+	// Initialize command
+	h.cmd = exec.CommandContext(ctx, args[0], args[1:]...)
+	h.cmd.Env = env
+	h.cmd.Dir = workDir
+
+	h.cmd.SysProcAttr = &syscall.SysProcAttr{
```

Comment:
Instead of an explicit syscall.Chroot call in the plugin handler, we are only chrooting the plugin process.